src/index.ts

```diff
@@ -39,62 +24,33 @@ export class ThemeResolverPlugin {
             this.pathRegex.push(new RegExp(`^${res.prefix}/`));
         });
         this.cache = {};
-        this.chosenResolver = {};
     }
 
     public apply(resolver: any) {
-        const target = resolver.ensureHook("module");
-
-        resolver.hooks.module.tapAsync(
-            "ThemeResolverPlugin",
-            (request: any, resolveContext: any, callback: () => void) => {
-                this.pathRegex.forEach((reg, x) => {
-                    if (request.request.match(reg)) {
-                        this.chosenResolver = Object.assign(ThemeResolverPlugin.defaultOptions, this.options[x]);
+        const target = resolver.ensureHook("resolved");
+
+        resolver.getHook("module")
+        .tapAsync("ThemeResolverPlugin", (request: any, resolveContext: any, callback: () => void) => {
```

Comment:
Can we somehow figure out what the types for `request` and `resolveContext` are?

Reply:
This should help: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/enhanced-resolve/index.d.ts

Reply:
Sadly the types are outdated. They are generated for version 3.0 and currently there is a version 5 in beta.